### ISSUE TITLE
Double gradient reduction in examples?

### ISSUE THREAD

**Issue by TimZaman** [kubeflow/pytorch-operator#122]
First, I'm confused why the example models implement their own DistributedDataParallel module. Why not use the torch one?

[DistributedDataParallel](https://pytorch.org/docs/master/nn.html#torch.nn.parallel.DistributedDataParallel) is used in the examples:
https://github.com/kubeflow/pytorch-operator/blob/master/examples/ddp/mnist/cpu/mnist_ddp_cpu.py#L154

The DistributedDataParallel documentation states:
>  During the backwards pass, gradients from each node are averaged.

However, a DIY `average_gradients` function is used in the same example _as well_: https://github.com/kubeflow/pytorch-operator/blob/master/examples/ddp/mnist/cpu/mnist_ddp_cpu.py#L168

Double trouble?

**Comment by andreyvelich** [comment]
Thank you for your issue.
Yes, we can use DistributedDataParallel from pytorch library. 
Will change it.
